plugin-server/src/llm-analytics/services/temporal.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Client, Connection, TLSConfig } from '@temporalio/client'
+import { Counter } from 'prom-client'
+
+import { Hub } from '../../types'
+import { logger } from '../../utils/logger'
+import { captureException } from '../../utils/posthog'
+
+const temporalWorkflowsStarted = new Counter({
+    name: 'evaluation_temporal_workflows_started',
+    help: 'Number of evaluation workflows started',
+    labelNames: ['status'],
+})
+
+export class TemporalService {
+    private client?: Client
+    private connecting?: Promise<Client>
+
+    constructor(private hub: Hub) {}
+
+    private async ensureConnected(): Promise<Client> {
+        if (this.client) {
+            return this.client
+        }
+
+        if (this.connecting) {
+            return await this.connecting
+        }
+
+        this.connecting = this.createClient()
+        this.client = await this.connecting
+        this.connecting = undefined
+
+        return this.client
+    }
+
+    private async createClient(): Promise<Client> {
+        // Configure TLS if certificates are provided (for production)
+        let tls: TLSConfig | boolean = false
+        if (this.hub.TEMPORAL_CLIENT_ROOT_CA && this.hub.TEMPORAL_CLIENT_CERT && this.hub.TEMPORAL_CLIENT_KEY) {
+            tls = {
+                serverRootCACertificate: Buffer.from(this.hub.TEMPORAL_CLIENT_ROOT_CA),
+                clientCertPair: {
+                    crt: Buffer.from(this.hub.TEMPORAL_CLIENT_CERT),
+                    key: Buffer.from(this.hub.TEMPORAL_CLIENT_KEY),
+                },
+            }
+        }
+
+        // Create connection first
+        const connection = await Connection.connect({
+            address: this.hub.TEMPORAL_HOST || 'localhost:7233',
+            tls,
+        })
+
+        // Then create client with connection
+        const client = new Client({
+            connection,
+            namespace: this.hub.TEMPORAL_NAMESPACE || 'default',
+        })
+
+        logger.info('✅ Connected to Temporal', {
+            address: this.hub.TEMPORAL_HOST,
+            namespace: this.hub.TEMPORAL_NAMESPACE,
+            tlsEnabled: tls !== false,
+        })
+
+        return client
+    }
+
+    async startEvaluationWorkflow(evaluationId: string, targetEventId: string): Promise<void> {
+        try {
+            const client = await this.ensureConnected()
+
+            const workflowId = `eval-${evaluationId}-${targetEventId}-${Date.now()}`
+
+            await client.workflow.start('run-evaluation', {
```

Comment:
suggestion: `client.workflow.start` method returns a handle that can be used to operate on the workflow (e.g. to cancel it, or await its result).

I would suggest returning that so that whoever calls `startEvaluationWorkflow` can eventually decide what to do with the handler. Otherwise, it gets lots and the caller would have to re-request the handler using the workflow id, which is also defined here and not returned.

I don't see this having an impact on this PR specifically, just a general suggestion!